image_services/glo_30_hand/make_hand_service.py

```diff
@@ -0,0 +1,173 @@
+import subprocess
+
+import arcpy
+
+
+dataset_name = 'GLO30_HAND'
+working_directory = '/home/arcgis/asjohnston/'
+raster_store = '/home/arcgis/raster_store/'
+
+geodatabase = f'{working_directory}{dataset_name}.gdb'
+mosaic_dataset = f'{geodatabase}/{dataset_name}'
+raster_function_template = f'{working_directory}GLO30_HAND_2SDstretch.rft.xml'
+overview_name = f'{dataset_name}_Overview'
+local_overview = f'/home/arcgis/raster_store/{overview_name}.crf'
+s3_overview = f'/vsis3/hyp3-nasa-disasters/overviews/{overview_name}.crf'
+service_definition_draft = f'{working_directory}{dataset_name}.sddraft'
+service_definition = f'{working_directory}{dataset_name}.sd'
+
+
+arcpy.env.parallelProcessingFactor = '75%'
+
+arcpy.management.CreateFileGDB(
+    out_folder_path=working_directory,
+    out_name=f'{dataset_name}.gdb',
+)
+
+arcpy.management.CreateMosaicDataset(
+    in_workspace=geodatabase,
+    in_mosaicdataset_name=dataset_name,
+    coordinate_system=3857,
+)
+
+arcpy.management.AddFields(
+    in_table=mosaic_dataset,
+    field_description=[
+        ['Tile', 'TEXT'],
+        ['DownloadURL', 'TEXT'],
+        ['URLDisplay', 'TEXT'],
+        ['Dataset_ID', 'TEXT'],
+    ],
+)
+
+arcpy.management.AddRastersToMosaicDataset(
+    in_mosaic_dataset=mosaic_dataset,
+    raster_type='Raster Dataset',
+    input_path='/vsis3/glo-30-hand/v1/2021/',
+    filter='REGEX:.*Copernicus_DSM_COG_10_[NS][0-8]\d_00_[EW]\d\d\d_00_HAND.tif',
+)
+
+arcpy.management.CalculateFields(
+    in_table=mosaic_dataset,
+    fields=[
+        ['Tile', '!Name!.split("_")[4] + !Name!.split("_")[6]'],
```

Comment:
What's this `!Name!` syntax in the string?

Reply:
The string on the right is an entire python expression to be evaluated. `!Name!` is a variable that contains the geotiff name without th3 `.tif`, e.g. `Copernicus_DSM_COG_10_N08_00_W065_00_HAND`

https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/calculate-fields.htm
https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/calculate-field-examples.htm

Reply:
> For Python calculations, field names must be enclosed in exclamation points (for example, !fieldname!).